doc/_index.md

```diff
@@ -1,13 +1,85 @@
 ---
 title: "API & Pipeline"
-date: 2022-03-04
-draft: true
-description: "KernelCI API and Modular Pipeline"
-weight: 4
+date: 2023-09-01
+description: "KernelCI API and Pipeline"
+weight: 2
 ---
 
-This section covers the new KernelCI API as well as the initial pipeline using
-it.
+## API Overview
+
+The KernelCI API is a server-side service which provides two main features: a
+database abstraction and a publisher / subscriber interface.  Another important
+concept is the fact that users own the data they send to the API.  Let's have a
+quick look at how this all fits together.
+
+### Database Abstraction
+
+All the data managed by KernelCI is stored in a MongoDB database using node
+objects.  These can contain data about any part of the testing hierarchy such
+as a kernel revision, a build, static test results, runtime functional tests,
+regressions etc.  Each node has a parent so they form a tree
+([DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph)).  There is
+typically one root node for each kernel revision with lots of child nodes
+containing all the test data that relates to it.
+
+Each node object also has a state which can be used when orchestrating the
+pipeline.  For example, a node will be in the Running state while awaiting some
+results.  There's also a result value, to tell whether the job passed or
+failed.  Finally there's a list of artifacts with URLs to know where to find
+all the related files (binaries, logs, generated results etc.).
+
+Here's a publicly available
+[graph](https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/public/dashboards/5b52c3c6-81bb-4658-a5f3-8ebd2e980436#)
+showing the number of nodes added to the database every day:
+
+<div style="padding: 0 0 20px 0"><iframe style="background: #FFFFFF;border: none;border-radius: 2px;box-shadow: 0 2px 10px 0 rgba(70, 76, 79, .2);" width="640" height="480" src="https://charts.mongodb.com/charts-kernelci-api-staging-otxuy/embed/charts?id=64c9fa16-ba81-40e9-8cd7-cd56fb88fbf3&maxDataAge=3600&theme=light&autoRefresh=true"></iframe></div>
+
+> **Note:** The API doesn't manage storage, the only requirement is to provide
+> publicly-available HTTP(s) URLs for each artifact.
```

Comment:
minor: typo: s/(s)/(S)/ ?

Reply:
Yes I think I've seen it written as HTTP(s) before but as it's either HTTP or HTTPS I think a capital letter makes more sense.